pallets/pallet-deposit-storage/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+
+mod deposit;
+pub mod traits;
+
+pub use deposit::FixedDepositCollectorViaDepositsPallet;
+pub use pallet::*;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use crate::{
+		deposit::{free_deposit, reserve_deposit, DepositEntry},
+		traits::DepositStorageHooks,
+	};
+
+	use super::*;
+
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{
+			fungible::{hold::Mutate, Inspect},
+			EnsureOrigin,
+		},
+	};
+	use frame_system::pallet_prelude::*;
+	use parity_scale_codec::FullCodec;
+	use scale_info::TypeInfo;
+	use sp_runtime::DispatchError;
+	use sp_std::fmt::Debug;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
```

Comment:
It's storage version 0, so migrations would not fail if they check that, right? The default would still be 0. I would migrate it once it is bumped to 1, no?

Reply:
Are you sure that the on-chain storage version is set to 0 by default for a new pallet? As far as I know, the on-chain version needs to be initialized. This is what we did in this [PR](https://github.com/KILTprotocol/kilt-node/commit/8e36fb26011887f726eaef21036cf56e02450c4a#diff-0fb85acd71505115281e23edf6e01a99b84b52eb8ba07629ba7518534a71d3d7R54) for the multisig pallet.

Reply:
No I'm not saying it is set to 0, but when being read, if not present, the default value of `0` is returned, which matches the pallet storage version, hence the check `stored_version == pallet_version` will not fail. Of course, once the pallet version is updated to `1`, then it will have to be written on storage. The issue with Substrate pallet, is that it was not written on storage when the pallet version was greater than `0`, and THAT is a problem.

Reply:
Ah, I see. There is a default implementation. Thanks for your explanation :racehorse: